template/registry_integer.go

```diff
@@ -8,20 +8,23 @@ import (
 
 // integerRegistryFunc is the implementation of the integer function. Locale-sensitive integer formatting.
 var integerRegistryFunc = RegistryFunc{
-	Format: integerFunc,
+	Format: integerFunc(Format),
```

Comment:
Would changing RegistryFunc simplify?

> Functions are used to evaluate, format, select, or otherwise process data values during formatting.

> A function's entry in the function registry will define whether the function is a selector or formatter (or both)...

What do you think?

```go
type FuncKind int

const (
	Formatter FuncKind = iota
	Selector
)

type RegistryFunc = func(operand any, options Options, locale language.Tag, kind FuncKind) (any, error)
```

Reply:
That would change the way currently supported functions are determined.

Reply:
I see. The current way is better for devex - e.g. custom functions don't need to handle kind.

Can we rename Match to Select to align naming with MF2 documentation?